nano/node/active_transactions.cpp

```diff
@@ -256,6 +280,34 @@ void nano::active_transactions::request_confirm (nano::unique_lock<std::mutex> &
 		bool const overflow_l (unconfirmed_count_l > node.config.active_elections_size && election_l->election_start < election_ttl_cutoff_l && !node.wallets.watcher->is_watched (i->root));
 		if (overflow_l || election_l->transition_time (solicitor))
 		{
+			if (election_l->failed () && election_l->optimistic ())
```

Comment:
We've had this function growing in size in the past and it would be nice to keep it tighter, can we extract something here?

Reply:
Extracted to new function